docs/overview/Lifecycle.md

```diff
@@ -0,0 +1,383 @@
+  * [Holocron Runtime](#holocron-runtime)
+* [Runtime App Configuration](#runtime-app-configuration)
+* [Server Side Rendering](#server-side-rendering)
+  * [Rendering Life Cycle](#rendering-life-cycle)
+  * [Partials Rendering](#partials-rendering)
+  * [Text Rendering](#text-rendering)
+
+## One App Server
+
+When the One App server first boots up, there are a series of steps
+undertaken before the server becomes operational - configuring One App
+based on environment settings is the first step, followed by pre-loading
+all the Holocron modules onto the server.
+
+### Server Settings
+
+This section will cover the main environment variables used by One App,
+and the how each can alter the way One App is ran.
+
+**Core Configuration**
+
+One App behaves differently depending on which node environment it is
+running under, thus whenever One App is ran, `NODE_ENV` must be defined
+with either `development` or `production`.
+
+>  * [`NODE_ENV`](../api/server/environment-variables#node_env) ⚠️
+
+One App has a uniform value used to indicate which environment the app is
+running in. `ONE_CONFIG_ENV` is used internally as well as usable in our
+modules (to render per environment).
+
+>  * [`ONE_CONFIG_ENV`](../api/server/environment-variables#one_config_env) ⚠️
+
+The One App micro-frontend architecture utilizes an entry point to render
+the entire app. This entry point is our root Holocron module and its name
+can be set with `ONE_CLIENT_ROOT_MODULE_NAME`. Without this value, One App
+would not know which module to load and render from the module map (discussed below).
+
+>  * [`ONE_CLIENT_ROOT_MODULE_NAME`](../api/server/environment-variables#one_client_root_module_name) ⚠️
+
+**Development Environment**
+
+In development mode, One App spins up a development CDN and API
+server that run on `localhost` which [`@americanexpress/one-app-runner`](one-app-runner)
+also utilizes. We can configure the ports to use for the
+development CDN and proxy servers ran alongside One App. By
+default, One App is ran on port `3000`, the [`@americanexpress/one-app-dev-cdn`](one-app-dev-cdn) development CDN
+is ran on port `3001` and the development proxy server [`@americanexpress/one-app-dev-proxy`](one-app-dev-proxy)
+is ran on port `3002`. To modify any of the three ports:
+
+>  * [`HTTP_PORT`](../api/server/environment-variables#http_port)
+>  * [`HTTP_ONE_APP_DEV_CDN_PORT`](../api/server/environment-variables#http_one_app_dev_cdn_port)
+>  * [`HTTP_ONE_APP_DEV_PROXY_SERVER_PORT`](../api/server/environment-variables#http_one_app_dev_proxy_server_port)
+
+There is a metrics server that runs in development and production,
+which tracks certain activities happening internally and reports the
+metrics on demand (during local development `http://localhost:3005/metrics`).
+The metrics server runs on port `3005` by default, to change the port:
+
+>  * [`HTTP_METRICS_PORT`](../api/server/environment-variables#http_metrics_port)
+
+**Production Environment**
```

Comment:
I would have production before development as everything that applied to production also mostly applies to development.

Reply:
That works too, I'll update.

Reply:
Updated (7e618c0)